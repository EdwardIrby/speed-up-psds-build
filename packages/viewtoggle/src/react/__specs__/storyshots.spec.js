import path from 'path'
import initStoryshots, {
  snapshotWithOptions
} from '@storybook/addon-storyshots'

function createNodeMock(element) {
  return { offsetLeft: 0, style: {} }
}

initStoryshots({
  configPath: path.resolve(__dirname, '..', '..', '..', '.storybook'),
  test: snapshotWithOptions({
    createNodeMock
  })
})
