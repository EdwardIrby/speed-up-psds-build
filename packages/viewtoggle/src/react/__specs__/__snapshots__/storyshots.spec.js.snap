// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots active default first 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        I'm active
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      I'm active
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots active explicit first 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        I'm active
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      I'm active
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots active explicit second 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        I'm active
      </span>
    </div>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      I'm active
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots active explicit third 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        I'm active
      </span>
    </div>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      I'm active
    </button>
  </div>
</div>
`;

exports[`Storyshots onSelect handles click 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        Option 1
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots options count more than max 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        Option 1
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots options count one 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        The Only Option
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      The Only Option
    </button>
  </div>
</div>
`;

exports[`Storyshots options count three 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        Option 1
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 3
    </button>
  </div>
</div>
`;

exports[`Storyshots options count two 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        Option 1
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      Option 1
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Option 2
    </button>
  </div>
</div>
`;

exports[`Storyshots text length all long 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        This one is really long because everyone likes those
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      This one is really long because everyone likes those
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      This one is really long because everyone likes those
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      This one is really long because everyone likes those
    </button>
  </div>
</div>
`;

exports[`Storyshots text length differing lengths 1`] = `
<div
  style={
    Object {
      "background": "#181818",
      "backgroundSize": "cover",
      "bottom": 0,
      "left": 0,
      "overflow": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
      "transition": "background 300ms",
    }
  }
>
  <div
    className="css-1ifazk6"
  >
    <div
      className="css-ztmayk"
      style={Object {}}
    >
      <span
        data-css-1jyyyen=""
      >
        This one is really long because everyone likes those
      </span>
    </div>
    <button
      aria-selected={true}
      className="css-1p9dg7s"
      onClick={[Function]}
    >
      This one is really long because everyone likes those
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      This is medium
    </button>
    <button
      aria-selected={false}
      className="css-1eee2k7"
      onClick={[Function]}
    >
      Short
    </button>
  </div>
</div>
`;
